abstractions/python/kiota/abstractions/serialization/serialization_writer.py

```diff
@@ -0,0 +1,186 @@
+from __future__ import annotations
+
+from abc import ABC
+from datetime import datetime
+from enum import Enum
+from io import BytesIO
+from typing import Any, Callable, Dict, List, Optional, TypeVar
+from uuid import UUID
+
+from .parsable import Parsable
+
+T = TypeVar("T")
+U = TypeVar("U", bound=Parsable)
+
+
+class SerializationWriter(ABC):
+    """Defines an interface for serialization of objects to a stream
+    """
+    def write_string_value(self, key: Optional[str], value: Optional[str]) -> None:
+        """Writes the specified string value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[str]): The string value to be written.
+        """
+        pass
+
+    def write_boolean_value(self, key: Optional[str], value: Optional[bool]) -> None:
+        """Writes the specified boolean value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[bool]): The boolean value to be written.
+        """
+        pass
+
+    def write_int_value(self, key: Optional[str], value: Optional[int]) -> None:
+        """Writes the specified integer value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[int]): The integer value to be written.
+        """
+        pass
+
+    def write_float_value(self, key: Optional[str], value: Optional[float]) -> None:
+        """Writes the specified float value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[float]): The float value to be written.
+        """
+        pass
+
+    def write_uuid_value(self, key: Optional[str], value: Optional[UUID]) -> None:
+        """Writes the specified uuid value to the stream with an optional given key.
+
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[UUId]): The uuid value to be written.
+        """
+        pass
+
+    def write_datetime_offset_value(self, key: Optional[str], value: Optional[datetime]) -> None:
```

Comment:
new date types

Reply:
Added in https://github.com/microsoft/kiota/pull/1058/commits/2b2dcec31d3448b6f1d9fda5b8a989675806916d